### PR TITLE
Feature/read line

### PR DESCRIPTION
**Pull request by BramProvoost**

### CODE REVIEW

File: srcs/main.c [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+{
+	t_token	*tokens;
+
+	if (!line)
+		return (EXIT_FAILURE);
+	tokens = tokenizer(line);
+	if (is_exit(tokens))
+		exit(ft_putendl_fd("exit", 1));
+
+	// tokens = tokenizer("''");	// fix empty quotes
```
Comment (Zangetsu89) [inline srcs/main.c:33]: All these comments are tests right? if yes then maybe it's better to make a separate test main file and clear this up

File: srcs/main.h [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+/* check_grammer.c */
+int	gramer_is_valid(t_token *tokens);
+
+/* signals.c */
+void	init_signals(void);
+
+#endif
+
+/* 
+What is the program flow for a lexer, tokenizer and parser?
```
Comment (Zangetsu89) [inline srcs/main.h:93]: All this commented out text can be moved into a readme file

File: srcs/tokenizer/token_printer.c [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	if (type == OUTPUT_D)
+		return ("OUTPUT_D");
+	return ("No tokens found");
+}
+
+void	print_tokens(t_token *tokens)
+{
+	while (tokens)
+	{
+		printf("type: %s\tlen: %li\ttext: \""GREEN"%s"NC"\"\n", type_to_string(tokens->type), tokens->len, tokens->value);
```
Comment (Zangetsu89) [inline srcs/tokenizer/token_printer.c:38]: We can't use printf right?

File: srcs/tokenizer/line_functions.c [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+/*   By: bprovoos <bprovoos@student.codam.nl>         +#+                     */
+/*                                                   +#+                      */
+/*   Created: 2023/01/26 16:51:25 by bprovoos      #+#    #+#                 */
+/*   Updated: 2023/02/03 10:58:23 by bprovoos      ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "../main.h"
+
+void	next_char(t_line *line)
```
Comment (Zangetsu89) [inline srcs/tokenizer/line_functions.c:15]: What's the reason this is in a function instead of just ++ whenever is used?

**Review by Zangetsu89 — CHANGES_REQUESTED** [review]
Small changes required, but all in all what you did seems very hard, so you gotta explain to me slowly. After making changes just merge to main please.

File: srcs/main.c (hunk 1 above)
Comment (BramProvoost) [reply srcs/main.c:33]: Yes, I wil make a separate test main for that

File: srcs/main.h (hunk 2 above)
Comment (BramProvoost) [reply srcs/main.h:93]: I agree

File: srcs/tokenizer/token_printer.c (hunk 3 above)
Comment (BramProvoost) [reply srcs/tokenizer/token_printer.c:38]: That is true, this function is temporary for visualizing the tokens

File: srcs/tokenizer/line_functions.c (hunk 4 above)
Comment (BramProvoost) [reply srcs/tokenizer/line_functions.c:15]: I was thinking for readability

**Review by Zangetsu89 — APPROVED** [review]

**Pull request merged**